router/handle.go

```diff
@@ -305,6 +305,17 @@ func (rt *Handle) commitStatusList(workerJobStatuses *[]workerJobStatus) {
 		if err != nil {
 			rt.logger.Error("Unmarshal of job parameters failed. ", string(workerJobStatus.job.Parameters))
 		}
+		errorCode, err := strconv.Atoi(workerJobStatus.status.ErrorCode)
+		if err != nil {
+			errorCode = 200
+		}
+		if rt.throttlerFactory != nil {
+			rt.throttlerFactory.Get(rt.destType, parameters.DestinationID).ResponseCodeReceived(errorCode) // send response code to throttler
+		} else {
+			rt.logger.Debugf(`[%v Router] :: ThrottlerFactory is nil. Not throttling destination with ID %s`,
+				rt.destType, parameters.DestinationID,
+			)
+		}
```

Comment:
I don't see any reason why throttlerFactory would be nil. Can we revise all relevant pieces of code?
```suggestion
		rt.throttlerFactory.Get(rt.destType, parameters.DestinationID).ResponseCodeReceived(errorCode) // send response code to throttler
```

Reply:
I had to add this because there were some tests that didn't have throttlerFactory when executing this part of the code

Reply:
Removed